vrs_anvil/evidence.py

```diff
@@ -1,207 +1,22 @@
-
-        # create dataframe from Terra data table
-        # https://github.com/broadinstitute/fiss/blob/master/firecloud/api.py
-        try:
-            response = fapi.get_entities_tsv(
-                os.environ["WORKSPACE_NAMESPACE"],
-                os.environ["WORKSPACE_NAME"],
-                "phenotype",
-                model="flexible",
-            )
-            response.raise_for_status()
-        except Exception as e:
-            if response.json() and e in response.json():
-                error_message = response.json()["message"]
-            else:
-                error_message = e
-            print(
-                f"Error while loading phenotype data table from workspace: \n{error_message}"
-            )
-
-        phenotype_tsv = io.StringIO(response.text)
-        phenotype_df = pd.read_csv(phenotype_tsv, sep="\t")
-    else:
-        # table path specified, parse using that table
-        with open(phenotype_table, "r") as file:
-            if phenotype_table.endswith(".csv"):
-                phenotype_df = pd.read_csv(file)
-            elif phenotype_table.endswith(".tsv"):
-                phenotype_df = pd.read_csv(file, sep="\t")
-            else:
-                raise Exception(
-                    "Only csv and tsv file types implemented for phenotype table"
-                )
-
-    # create patient to phenotype dict
-    phenotype_index = {}
-    for participant_id in phenotype_df["participant_id"].unique():
-        phenotypes = phenotype_df[phenotype_df["participant_id"] == participant_id][
-            "term_id"
-        ].unique()
-
-        phenotype_index[participant_id] = (
-            set(phenotypes) if as_set else list(phenotypes)
-        )
-
-    # save to disk if specified
-    if save_path:
-        if os.path.exists(save_path):
-            raise Exception(
-                f"index already exists at path: {save_path}. Please delete it before continuing"
-            )
-        else:
-            with open(save_path, "w") as file:
-
-                # make it serializable so can be written to disk
-                if as_set:
-                    for patient, pheno_set in phenotype_index.items():
-                        phenotype_index[patient] = list(pheno_set)
-
-                json.dump(phenotype_index, file)
-
-    return phenotype_index
-
-
 def get_cohort_allele_frequency(
     variant_id: str,
     vcf_path: str,
-    vcf_index_path: str = None,
-    phenotype_table: str = None,
-    phenotype_index_path: str = None,
-    participant_list: list[str] = None,
-    phenotype: str = None,
-    plugin_name: str = "GregorPlugin",
+    vcf_index_path: str | None = None,
+    participant_list: list[str] | None = None,
+    phenotype: str | None = None,
+    plugin: BasePlugin = BasePlugin(),
```

Comment:
It shouldn't matter here, but I suspect a linter would get angry about this use of a class constructor function as a default arg given Python's weirdness about eagerly evaluating defaults. It might be better to call this `BasePlugin | None` and handle the default case in the function just to be safe